File: caesar.c
#include <cs50.h>
#include <stdio.h>
#include <ctype.h>
#include <string.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    if(argc != 2)
    {
        printf("not desired count of arguments\n");
        return 1;
    }
    int k = atoi(argv[1]);

    if(k < 0)
    {
        printf("can't be a negative number\n");
        return 1;
    }
    else
    {

        printf("plaintext: ");
        string p = get_string();
        printf("ciphertext: ");

        for(int i = 0; i < strlen(p); i++)
        {
            if(isalpha(p[i]))
            {
                if(isupper(p[i]))
                {
                    printf("%c", (((p[i] + k) - 65) % 26) + 65);
                }
                else if(islower(p[i]))
                {
                    printf("%c", (((p[i] + k) - 97) % 26) + 97);

                }
                else
                {
                    printf("%c", p[i]);
                }
            }
        }
    }
    printf("\n");
    return 0;
}

File: caesar.py
from cs50 import get_string
import sys


def main():
    # get correct # of c-l arguments
    if len(sys.argv) != 2:
        print("not desired count of arguments")
        exit(1)

    # convert key from string to integer
    k = int(sys.argv[1])
    if k < 0:
        print("can't be a negative number!")
        exit(1)

    # get plaintext from user
    p = get_string("plaintext: ")
    print("ciphertext: ", end="")

    # print out ciphertext
    for c in p:
        if c.isalpha():
            if c.isupper():
                n = ((((ord(c) + k) - 65) % 26) + 65)
                print(f"{chr(n)}", end="")
            else:
                n = ((((ord(c) + k) - 97) % 26) + 97)
                print(f"{chr(n)}", end="")
        else:
            print(f"{c}", end="")

    print()
    exit(0)


if __name__ == "__main__":
    main()
